solidity/test/DepositUtilsTest.js

```diff
@@ -641,6 +642,71 @@ describe("DepositUtils", async function() {
       )
     })
   })
+  describe("enableWithdrawal()", async () => {
+    beforeEach(async () => {
+      await createSnapshot()
+    })
+
+    afterEach(async () => {
+      await restoreSnapshot()
+    })
+
+    it("correctly adds value to withdrawalAllowances", async () => {
+      const value = new BN(10000)
+      const beneficiary = accounts[1]
+
+      await testDeposit.enableWithdrawal(accounts[1], value)
+
+      const withdrawable = await testDeposit.getWithdrawAllowance.call({
+        from: beneficiary,
+      })
+
+      expect(withdrawable).to.eq.BN(value)
+    })
+  })
+
+  describe("withdrawFunds()", async () => {
+    beforeEach(async () => {
+      await createSnapshot()
+    })
+
+    afterEach(async () => {
+      await restoreSnapshot()
+    })
+
+    it("Correctly withdraws withdrawable balance at end-state", async () => {
+      await testDeposit.setState(states.LIQUIDATED)
+      const value = new BN(web3.utils.toWei("0.1"))
+      const beneficiary = accounts[1]
+      const initialBalance = await web3.eth.getBalance(beneficiary)
+
+      await ecdsaKeepStub.pushFundsFromKeep(testDeposit.address, {value: value})
+      await testDeposit.enableWithdrawal(beneficiary, value)
+      await testDeposit.withdrawFunds({from: beneficiary})
+
+      const withdrawable = await testDeposit.getWithdrawAllowance.call({
+        from: beneficiary,
+      })
+      const finalBalance = await web3.eth.getBalance(beneficiary)
+
+      expect(withdrawable).to.eq.BN(0)
+      expect(finalBalance).to.gt.BN(initialBalance)
```

Comment:
Just because the final balance is > initial balance, doesn't mean we succeeded, right? We need to make sure we withdraw initial balance + the value enabled for withdrawal.

Reply:
That's right, I got lazy accounting for the gas cost since we're calling from `beneficiary`. Meant to fix this